### ISSUE TITLE
ListView items below SliverAppBar when scrolled

### ISSUE THREAD

**Issue by mounthorse-slns** [flutter/flutter#27906]
Hi, I'm using `NestedScrollView` with `SliverAppBar` and a simple `ListView` to show contents.
Here is my code:

**main.dart**

```dart
    import 'package:flutter/material.dart';
    import 'package:flutter_app/item.dart';
    import 'package:flutter_app/item_listing.dart';
    import 'package:transparent_image/transparent_image.dart';
    
    void main() => runApp(MyApp());
    
    class MyApp extends StatelessWidget {
      @override
      Widget build(BuildContext context) {
        return MaterialApp(
          title: 'Flutter Demo',
          theme: ThemeData(
            primarySwatch: Colors.blue,
          ),
          home: MyHomePage(title: 'Flutter Demo Home Page'),
        );
      }
    }
    
    class MyHomePage extends StatefulWidget {
      MyHomePage({Key key, this.title}) : super(key: key);
    
      final String title;
    
      @override
      _MyHomePageState createState() => _MyHomePageState();
    }
    
    class _MyHomePageState extends State<MyHomePage> {
      static const double kExpandedHeight = 200.0;
    
      List<Item> items = [
        Item('Item0', 'Item 0 desc'),
        Item('Item1', 'Item 1 desc'),
        Item('Item2', 'Item 2 desc'),
        Item('Item3', 'Item 3 desc'),
      ];
    
      @override
      Widget build(BuildContext context) {
        return Scaffold(
          floatingActionButton: FloatingActionButton(
            onPressed: () {
                print('NAVIGATE TO ADD');
            },
            tooltip: 'Add item',
            child: Icon(Icons.add),
          ),
          body: NestedScrollView(
            headerSliverBuilder: (BuildContext context, bool innerBoxIsScrolled) {
              return <Widget>[
                SliverAppBar(
                  expandedHeight: kExpandedHeight,
                  pinned: true,
                  floating: false,
                  snap: false,
                  forceElevated: true,
                  flexibleSpace: FlexibleSpaceBar(
                    centerTitle: true,
                    title: Row(
                        mainAxisAlignment: MainAxisAlignment.center,
                        mainAxisSize: MainAxisSize.max,
                        children: <Widget>[
                          Text('Items'),
                        ]
                    ),
                    background: FadeInImage.memoryNetwork(
                      placeholder: kTransparentImage,
                      image: 'http://www.reset-informatica.it/wp-content/uploads/2015/03/header_1.png',
                      fit: BoxFit.cover,
                    ),
                  ),
                  iconTheme: IconThemeData(
                    color: Colors.white,
                  ),
                ),
              ];
            },
            body: _buildBody(),
          ),
        );
      }
    
      Widget _buildBody() {
        return Container(
          color: Colors.white,
          child: ItemListing(
            items: items,
            onItemTap: (item) {
              print('NAVIGATE TO DETAILS');
            },
            onItemLongPress: (item) {
              print('NAVIGATE TO EDIT');
            },
          ),
        );
      }
    }

```

**item_listing.dart**

```dart
import 'package:flutter/material.dart';
import 'package:meta/meta.dart';

    class ItemListing extends StatelessWidget {
      ItemListing({
        @required this.items,
        @required this.onItemTap,
        this.onItemLongPress,
        this.selectedItem
      });
    
      final List<Object> items;
      final ValueChanged<Object> onItemTap;
      final ValueChanged<Object> onItemLongPress;
      final dynamic selectedItem;
    
      final ScrollController _scrollController = new ScrollController();
    
      @override
      Widget build(BuildContext context) {
        return items.isEmpty
            ? Center(
                child: Text('No items.', textAlign: TextAlign.center, style: TextStyle(fontSize: 18.0, fontStyle: FontStyle.italic)),
              )
            : ListView.builder(
                shrinkWrap: false,
                controller: _scrollController,
                itemCount: items.length,
                itemBuilder: (context, index) =>
                    _buildListItem(context, items[index], index),
              );
      }
    
      Widget _buildListItem(BuildContext context, dynamic data, int index) {
        return Padding(
          key: ValueKey(data.name),
          padding: const EdgeInsets.symmetric(horizontal: 16.0),
          child: Card(
            elevation: 4.0,
            child: ListTile(
              isThreeLine: false,
              title: Text(data.name, style: TextStyle(fontSize: 20.0)),
              // subtitle: Text(_getDateForDisplay(route.createdOn)),
              subtitle: Column(
                mainAxisAlignment: MainAxisAlignment.start,
                crossAxisAlignment: CrossAxisAlignment.start,
                children: <Widget>[
                  Text(data.description),
                  Text('Created by User'),
                ],
              ),
              trailing: Text('#$index', style: TextStyle(fontSize: 20.0, fontWeight: FontWeight.bold)),
              onTap: () => onItemTap(data),
              onLongPress: () => onItemLongPress != null ? onItemLongPress(data) : {},
              selected: selectedItem?.name == data.name,
            ),
          ),
        );
      }
    }
```

**item.dart**

```dart
    class Item {
      String name;
      String description;
    
      Item(String name, String description) {
        this.name = name;
        this.description = description;
      }
    }
```

**pubspec.yaml**

    [...]
    dependencies:
      flutter:
        sdk: flutter
    
      # The following adds the Cupertino Icons font to your application.
      # Use with the CupertinoIcons class for iOS style icons.
      cupertino_icons: ^0.1.2
      transparent_image: ^0.1.0
    [...]

The problem is that even with one item the content scrolls and that item go behind the AppBar:

<img src="https://user-images.githubusercontent.com/19502861/52751524-f542dd80-2fef-11e9-92c1-257595991209.jpg" height="500">

I would like to achieve that instead:

<img src="https://user-images.githubusercontent.com/19502861/52751549-0e4b8e80-2ff0-11e9-8f75-8f898b255138.jpg" height="500">

What am I missing? I need to use `ListView.builder` or `SliverList` instead of `ListView` or am I missing some configuration?

**Flutter Analyze**

```
Analyzing moon_board_manager...

0 issues found. (ran in 2.5s)
```

**Flutter Doctor**

```
[√] Flutter (Channel stable, v1.0.0, on Microsoft Windows [Versione 10.0.17134.590], locale it-IT)
    • Flutter version 1.0.0 at E:\Programmazione\Libraries\flutter
    • Framework revision 5391447fae (3 months ago), 2018-11-29 19:41:26 -0800
    • Engine revision 7375a0f414
    • Dart version 2.1.0 (build 2.1.0-dev.9.4 f9ebf21297)

[√] Android toolchain - develop for Android devices (Android SDK 28.0.3)
    • Android SDK at E:\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = E:\Android\sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)
    • All Android licenses accepted.

[√] Android Studio (version 3.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 32.0.1
    • Dart plugin version 182.5215
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)

[√] VS Code, 32-bit edition
    • VS Code at C:\Program Files (x86)\Microsoft VS Code
    • Flutter extension version 2.22.3

[√] VS Code, 64-bit edition (version 1.31.0)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 2.22.3

[!] Connected device
    ! No devices available

! Doctor found issues in 1 category.
```

**Comment by Kartik1607** [comment]
Please post *minimal* code (Such that your code could be copy pasted and it would run). Your current code is too much verbose and needs a lot of editing to make it work.

**Comment by bm-software** [comment]
@Kartik1607 Ok, you're right. I've updated the first post with minimal code to reproduce the problem. I've not capture screens again (so the layout is different) but the problem is the same.

**Comment by yanshaokun** [comment]
I have the same problem
It seems that the controller in the NestedScrollView conflicts with the controller in the ListView

**Comment by adesugbaa** [comment]
I am interested in this problem as well. My list view scrolls under the SliverAppBar when collapsed.

**Comment by sprabhdeep** [comment]
I am also need to achieve this. Want to pin the outer scrollview below the tabbar and inner scrollview could scroll under them. Any suggestions?

**Comment by vanlooverenkoen** [comment]
Someone who already found a solution to this problem?

**Comment by vanlooverenkoen** [comment]
Add a `SliverOverlapAbsorber` 
the handle attribute will fix the issue

```
            NestedScrollView(
              headerSliverBuilder: (context, innerBoxIsScrolled) => [
                    SliverOverlapAbsorber(
                      child: SliverAppBar(
                        expandedHeight: 200,
                        flexibleSpace: FlexibleSpaceBar(
                            title: Text('TITLE'),
                            background: Container(color: Colors.amber)
                      ),
                      handle: NestedScrollView.sliverOverlapAbsorberHandleFor(context),
                    ),
                  ],
              body: _buildYourList(),
            );
```

**Comment by mounthorse-slns** [comment]
@vanlooverenkoen Thank you! Your solution fixes the issue! :)

**Comment by TheJulianJES** [comment]
Although this is not the exact issue, this fix doesn't really work with a RefreshIndicator, as it would now appear in the SliverAppBar using the code provided.
But thanks @vanlooverenkoen!

**Comment by vanlooverenkoen** [comment]
@TheJulianJES you should create a new issue because this should be tracker seperatly. I will implement a refreshlayout tomorrow

**Comment by wemped** [comment]
@vanlooverenkoen is there another issue for refreshlayout?

I'm having trouble with having a floating `SliverAppBar`, it compresses just fine when you scroll down, but it does not expand back to normal when you scroll back up - at least not until you scroll back to the top..

**Comment by vanlooverenkoen** [comment]
Not yet, feel free to create one

**Comment by wemped** [comment]
@vanlooverenkoen 👍 done, https://github.com/flutter/flutter/issues/34727

**Comment by Git-Gabo** [comment]
> @vanlooverenkoen is there another issue for with a RefreshIndicator?
> 
> I'm having trouble with having a floating `SliverAppBar`, it compresses just fine when you scroll down, but it does not expand back to normal when you scroll back up - at least not until you scroll back to the top..

Maybe adding the "floating" parameter and setting it to *true* will solve the problem.

**Comment by akshg05** [comment]
How can we resolve this issue with multiple sliver headers? The framework says only one 
"A SliverOverlapAbsorberHandle cannot be passed to multiple RenderSliverOverlapAbsorber objects at the same time."
